### PR TITLE
Merge wpcom/11may2015

### PR DESCRIPTION
**Pull request by dereksmart**
Clean up unnecessary REST API tests
Update GlotPress - more languages 
New Dailymotion shortcode

### CODE REVIEW

File: json-endpoints/class.wpcom-json-api-site-settings-endpoint.php [hunk 1]
```diff
@@ -190,7 +196,7 @@ public function update_settings() {
 
 		// $this->input() retrieves posted arguments whitelisted and casted to the $request_format
 		// specs that get passed in when this class is instantiated
-		$input = $this->input();
+		$input = apply_filters( 'rest_api_update_site_settings', $this->input() );
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-site-settings-endpoint.php]: We will need to add a docblock here.

File: modules/publicize/publicize.php [hunk 2]
```diff
@@ -97,18 +97,15 @@ function get_profile_link( $service_name, $c ) {
 		$cmeta = $this->get_connection_meta( $c );
 
 		if ( isset( $cmeta['connection_data']['meta']['link'] ) ) {
-			if ( 'facebook' == $service_name && 0 === strpos( parse_url( $cmeta['connection_data']['meta']['link'], PHP_URL_PATH ), '/app_scoped_user_id/' ) ) {
```
Comment (jeherve) [inline modules/publicize/publicize.php]: This merge seems to overwrite the changes made by @aduth in #1944

File: modules/shortcodes/facebook.php [hunk 3]
```diff
@@ -43,16 +39,11 @@ function jetpack_facebook_embed_handler( $matches, $attr, $url ) {
 	if ( defined( 'DOING_AJAX' ) && DOING_AJAX && ! empty( $_POST['action'] ) && 'parse-embed' == $_POST['action'] ) {
 		return $embed . '<script src="//connect.facebook.net/en_US/all.js#xfbml=1"></script>';
 	} else {
+		wp_enqueue_script( 'jetpack-facebook-embed', plugins_url( 'js/facebook.js', __FILE__ ), array( 'jquery' ) );
```
Comment (jeherve) [inline modules/shortcodes/facebook.php]: That file wasn't ported to Jetpack, it needs to be included in the build script

File: modules/shortcodes/instagram.php [hunk 4]
```diff
@@ -139,40 +138,36 @@ function jetpack_instagram_handler( $matches, $atts, $url ) {
 	}
 
 	if ( ! empty( $response_body->html ) ) {
-		if ( ! $did_script ) {
-			$did_script = true;
-			add_action( 'wp_footer', 'jetpack_instagram_add_script' );
-		}
-
-		// there's a script in the response, which we strip on purpose since it's added above
+		wp_enqueue_script( 'jetpack-instagram-embed', plugins_url( 'js/instagram.js', __FILE__ ), array( 'jquery' ), false, true );
```
Comment (jeherve) [inline modules/shortcodes/instagram.php]: That file wasn't ported to Jetpack, it needs to be included in the build script

**Comment by jeherve** [comment]
@dereksmart I made a few comments above ^

In the future, shouldn't we follow the same process as we do for other PRs, and have another developer review the commits before merging?

File: json-endpoints/class.wpcom-json-api-site-settings-endpoint.php (hunk 1 above)
Comment (dereksmart) [inline json-endpoints/class.wpcom-json-api-site-settings-endpoint.php]: @jkudish when you get a minute would you mind ^ ?

File: modules/shortcodes/facebook.php (hunk 3 above)
Comment (dereksmart) [inline modules/shortcodes/facebook.php]: File added in cbc53d4, adding to build script now.

File: modules/shortcodes/instagram.php (hunk 4 above)
Comment (dereksmart) [inline modules/shortcodes/instagram.php]: File added in cbc53d4, adding to build script now.

File: modules/publicize/publicize.php (hunk 2 above)
Comment (dereksmart) [inline modules/publicize/publicize.php]: Thanks - missed this one, was meaning to strip it out along with the other Facebook changes and ask @aduth what is/isn't safe to have in our build script anymore.

Comment (aduth) [inline modules/publicize/publicize.php]: The changes in #1944 are the preferred version in both environments. I'll plan on updating this in WPCOM later today. Thanks for catching this @jeherve.

File: json-endpoints/class.wpcom-json-api-site-settings-endpoint.php (hunk 1 above)
Comment (jkudish) [inline json-endpoints/class.wpcom-json-api-site-settings-endpoint.php]: delegating to @aduth since that was his work ;)

Comment (aduth) [inline json-endpoints/class.wpcom-json-api-site-settings-endpoint.php]: See #2143 (merged)

**Pull request merged**